From 8a4ecffccd2537bdb91e728d1603ca792378e2e4 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Mon, 21 Aug 2023 17:51:47 +0200
Subject: [PATCH 327/954] refactor(st-clock): data size optimization

Reduce type of gate_refcounts[], uint8_t is enough.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 4172452d83d5718228c20534552d34bb32b5c792 refactor(st-clock): data size optimization
356a9edc2d3edeef5afe7aa49bfad848f21f6b04 refactor(st-clock): data size optimization]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I336455bcfd5a46b2a59cdbb9c78dce7ffc12c1a9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/323780
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/st/clk/clk-stm32-core.h | 2 +-
 drivers/st/clk/clk-stm32mp13.c  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/st/clk/clk-stm32-core.h b/drivers/st/clk/clk-stm32-core.h
index 285120dbc4..1cf9fa426e 100644
--- a/drivers/st/clk/clk-stm32-core.h
+++ b/drivers/st/clk/clk-stm32-core.h
@@ -73,7 +73,7 @@ struct stm32_clk_priv {
 	const uint32_t nb_div;
 	struct clk_oscillator_data *osci_data;
 	const uint32_t nb_osci_data;
-	uint32_t *gate_refcounts;
+	uint8_t *gate_refcounts;
 	void *pdata;
 };
 
diff --git a/drivers/st/clk/clk-stm32mp13.c b/drivers/st/clk/clk-stm32mp13.c
index ddad4ad778..b37ee2a5a3 100644
--- a/drivers/st/clk/clk-stm32mp13.c
+++ b/drivers/st/clk/clk-stm32mp13.c
@@ -889,7 +889,7 @@ static bool pll4_bootrom;
 #endif
 
 /* RCC clock device driver private */
-static unsigned int refcounts_mp13[CK_LAST];
+static uint8_t refcounts_mp13[CK_LAST];
 
 static const struct stm32_clk_pll *clk_st32_pll_data(unsigned int idx);
 
-- 
2.53.0

